data_classes_test.py

```diff
@@ -0,0 +1,60 @@
+"""
+Unit tests for data classes
+
+Implemented:
+    Data
+
+TODO:
+    DataController
+    Dataset
+    Portfolio
+"""
+
+import unittest
+from datetime import datetime
+from thousandaire.data_classes import Data
+
+class TestData(unittest.TestCase):
+    """
+    Unit test object for Data.
+    """
+    def setUp(self):
+        self.data = Data('test', ['buy', 'sell'])
+
+    def test_fields(self):
+        """
+        Test fields.
+        """
+        self.assertEqual(self.data.fields, ['date', 'buy', 'sell'])
+
+    def test_append(self):
+        """
+        Test append method in Data.
+        """
+        append_datum = (datetime(2020, 1, 27), 30, 31)
+        # Normal append.
+        self.data.append(append_datum)
+        self.assertEqual(tuple(self.data[0]), append_datum)
+        # Invalid append.
+        self.assertRaises(TypeError, self.data.append, 1)
+        self.assertRaises(TypeError, self.data.append, (3, 5))
+        self.assertRaises(TypeError, self.data.append, *append_datum)
+
+    def test_extend(self):
+        """
+        Test extend method in Data.
+        """
+        data_list = [
+            (datetime(2020, 1, 22), 23, 22),
+            (datetime(2020, 1, 23), 23, 22),
+            (datetime(2020, 1, 24), 23, 22)]
+        # Normal extend
+        self.data.extend(data_list)
+        self.assertEqual([tuple(item) for item in self.data], data_list)
```

Comment:
Can you extend twice (with different values) and so that we can make sure it works whenever it is empty or not?

Besides, can you validate items by item.(field) instead of converting items to tuples?
Such tests can't confirm that they are namedtuple.